scrape/scrape.go

```diff
@@ -414,21 +425,41 @@ func (sp *scrapePool) Sync(tgs []*targetgroup.Group) {
 
 	sp.targetMtx.Lock()
 	var all []*Target
-	sp.droppedTargets = []*Target{}
+
+	// Clean up old targets from drop cache.
+	for k, t := range sp.droppedTargets {
+		if time.Since(t.dropTime) > droppedTargetTTL {
```

Comment:
There's no need for the complication of a TTL, and it's not as if once a target disappears we'd expect it to come back. It's also be incorrect, as we'd be returning targets that are no longer being returned by SD.

Build up a new droppedTargets each time.

Reply:
K8S produces many new targets and if we won't delete the old ones we'll end up with an overflow of this map. Cronjobs may spam this map easily for example.

Reply:
I can't build this map each time because to build I should run `targetsFromGroup`, relabeling, etc

Reply:
You can build it each time, if but you don't need to run targetsFromGroup on something that came from SD and that was in the old map.